cdap-security/src/main/java/co/cask/cdap/security/guice/SecureStoreModules.java

```diff
@@ -19,33 +19,53 @@
 import co.cask.cdap.api.security.store.SecureStore;
 import co.cask.cdap.api.security.store.SecureStoreManager;
 import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.conf.SConfiguration;
 import co.cask.cdap.common.runtime.RuntimeModule;
-import co.cask.cdap.security.store.DummyKMSStore;
+import co.cask.cdap.security.store.DummySecureStore;
 import co.cask.cdap.security.store.FileSecureStore;
-import co.cask.cdap.security.store.SecureStoreUtils;
+import com.google.common.base.Strings;
 import com.google.inject.AbstractModule;
 import com.google.inject.Inject;
 import com.google.inject.Injector;
 import com.google.inject.Module;
 import com.google.inject.Provider;
 import com.google.inject.Singleton;
 import com.google.inject.TypeLiteral;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
  * Guice bindings for security store related classes.
+ *
+ * file - InMemory    - with password       - Valid, returns FileSecureStore
+ *                    - without password    - Invalid, throws IllegalArgumentException. (Set password)
+ *      - Standalone  - with password       - Valid, returns FileSecureStore
+ *                    - without password    - Invalid, throws IllegalArgumentException. (Set password)
+ *      - Distributed - with password       - Invalid, throws IllegalArgumentException. (mode not supported)
+ *                    - without password    - Invalid, throws IllegalArgumentException. (mode not supported)
+ *
+ * kms  - InMemory    - > Hadoop 2.6        - Invalid, throws IllegalArgumentException. (mode not supported)
```

Comment:
It won’t. Mini KMS is only for unit testing.


> On Aug 2, 2016, at 3:31 PM, Bhooshan Mogal <notifications@github.com> wrote:
> 
> In cdap-security/src/main/java/co/cask/cdap/security/guice/SecureStoreModules.java <https://github.com/caskdata/cdap/pull/6293#discussion_r73250541>:
> 
> >  
> >  /**
> >   * Guice bindings for security store related classes.
> > + *
> > + * file - InMemory    - with password       - Valid, returns FileSecureStore
> > + *                    - without password    - Invalid, throws IllegalArgumentException. (Set password)
> > + *      - Standalone  - with password       - Valid, returns FileSecureStore
> > + *                    - without password    - Invalid, throws IllegalArgumentException. (Set password)
> > + *      - Distributed - with password       - Invalid, throws IllegalArgumentException. (mode not supported)
> > + *                    - without password    - Invalid, throws IllegalArgumentException. (mode not supported)
> > + *
> > + * kms  - InMemory    - > Hadoop 2.6        - Invalid, throws IllegalArgumentException. (mode not supported)
> this will change when we have mini-kms, no? Perhaps add a todo with a jira
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/caskdata/cdap/pull/6293/files/7214cc401842ba8ee0980ad47dea81a2c146ba6e#r73250541>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAFGk5UFA_hUV3QOBg5HUKlZwBQ32Hd3ks5qb8U1gaJpZM4Ja37Y>.
>